src/Parser/Priest/Discipline/Modules/Spells/Contrition.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatPercentage, formatNumber } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+
+import Penance from '../Spells/Penance';
+import { OffensivePenanceBoltEstimation } from '../../SpellCalculations';
+
+class Contrition extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+    penance: Penance, // we need this to add `penanceBoltNumber` to the damage and heal events
+  };
+
+  healing = 0;
+  damagePenalty = 0;
+  player = null;
+  penanceBoltEstimation;
+
+  on_initialized() {
+    this.active = this.owner.modules.combatants.selected.hasTalent(SPELLS.CONTRITION_TALENT.id);
+    this.penanceBoltEstimation = OffensivePenanceBoltEstimation(this.statTracker);
+  }
+
+  /**
+   * Theory behind this is as follows
+   * By casting a contrition penance, you lose an offensive penance worth of healing
+   * To hone in on the true value of contrition, we estimate the healing of an offensive
+   * penance hit, and subtract that from the contrition amount.
+   *
+   * We keep the penance heal as that is a true gain from choosing a contrition
+   * penance over a regular offensive one.
+   */
+  on_byPlayer_heal(event) {
+    if (event.ability.guid !== SPELLS.CONTRITION_HEAL.id && event.ability.guid !== SPELLS.PENANCE_HEAL.id) {
+      return;
+    }
+
+    // Add the healing to our count
+    this.healing += event.amount;
+
+    // Get an estimated amount of damage and healing for a bolt
+    const { damage, healing } = this.penanceBoltEstimation();
```

Comment:
iirc Power of the Dark side was added baseline recently on Beta, so might need to account for that.

Reply:
Yeah, need to decide on this at some stage. Thinking just adding a buff tracker.